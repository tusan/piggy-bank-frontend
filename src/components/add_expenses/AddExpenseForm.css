.expenses-form {
  padding: 10px;
}

.ant-form-vertical .ant-form-item {
  padding: 0px;
}

.ant-form-item {
  margin-bottom: 10px;
}

.ant-calendar-picker {
  width: 100%;
}

.add-expense-form .ant-btn-primary {
  width: 100%;
}

.ant-form-explain {
  float: left;
}

@media screen and (min-width: 1224px) {
  .add-expense-form {
    border: 1px solid;
    border-color: #e8e8e8;
    margin-top: 20px;
    border-radius: 5px;
  }

  .add-expense-form h3 {
    color: rgba(0, 0, 0, 0.65);
  }
}
