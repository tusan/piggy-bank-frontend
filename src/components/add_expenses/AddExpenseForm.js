import React, { PureComponent } from 'react';
import { Form, DatePicker, Input, Button, Select } from 'antd';
import './AddExpenseForm.css';
import { InputNumber } from 'antd';

const { Item } = Form;
const { Option } = Select;
const { TextArea } = Input;

class AddExpenseForm extends PureComponent {
  handleSubmit = e => {
    e.preventDefault();

    this.props.form.validateFields((err, values) => {
      if (!err) {
        this.props.onSubmitSuccess(values);
      }
    });
  };

  render = () => {
    const { getFieldDecorator } = this.props.form;
    return (
      <div className="add-expense-form">
        <h3 className="form-title">Add New</h3>
        <Form
          layout="vertical"
          className="expenses-form"
          onSubmit={this.handleSubmit}
        >
          <Item>
            {getFieldDecorator('date', {
              rules: [
                {
                  required: true,
                  message: 'Please insert a valid date'
                }
              ]
            })(
              <DatePicker
                className="expense-date-picker"
                format={'DD/MM/YYYY'}
              />
            )}
          </Item>
          <Item>
            {getFieldDecorator('type', {
              rules: [
                {
                  required: true,
                  message: 'Please select a valid product'
                }
              ]
            })(
              <Select
                showSearch
                optionFilterProp="children"
                placeholder="Select type"
              >
                <Option value="MOTO">Moto</Option>
                <Option value="CASA">Casa</Option>
                <Option value="VARIE">Varie</Option>
              </Select>
            )}
          </Item>
          <Item>
            {getFieldDecorator('amount', {
              rules: [
                {
                  required: true,
                  message: 'Please insert a valid amount'
                }
              ]
            })(<InputNumber placeholder={'Insert amount: â‚¬'} />)}
          </Item>
          <Item>
            {getFieldDecorator('description', {})(
              <TextArea placeholder={'Insert short description'} />
            )}
          </Item>
          <Item>
            <Button type="primary" htmlType="submit">
              Add
            </Button>
          </Item>
        </Form>
      </div>
    );
  };
}

export default Form.create()(AddExpenseForm);
