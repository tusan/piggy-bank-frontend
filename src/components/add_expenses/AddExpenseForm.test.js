import React from 'react';
import renderer from 'react-test-renderer';
import AddExpenseForm from './AddExpenseForm';

describe('<AddExpenseForm />', () => {
  it('should display an AddExpenseForm', () => {
    const form = renderer.create(<AddExpenseForm />);

    expect(form).toMatchSnapshot();
  });
});
