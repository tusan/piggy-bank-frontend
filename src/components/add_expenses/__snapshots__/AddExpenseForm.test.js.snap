// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddExpenseForm /> should display an AddExpenseForm 1`] = `
<div
  className="add-expense-form"
>
  <h3
    className="form-title"
  >
    Add New
  </h3>
  <form
    className="ant-form ant-form-vertical expenses-form"
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <span
              className="expense-date-picker ant-calendar-picker"
              id="date"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            >
              <div
                onClick={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
              >
                <input
                  className="ant-calendar-picker-input ant-input"
                  placeholder="Select date"
                  readOnly={true}
                  value=""
                />
                <i
                  className="anticon anticon-calendar ant-calendar-picker-icon"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="calendar"
                    fill="currentColor"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                    />
                  </svg>
                </i>
              </div>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <div
              className="ant-select ant-select-enabled"
              id="type"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseOut={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
            >
              <div
                aria-autocomplete="list"
                aria-controls="test-uuid"
                aria-expanded={false}
                aria-haspopup="true"
                className="ant-select-selection
            ant-select-selection--single"
                data-__field={
                  Object {
                    "name": "type",
                  }
                }
                data-__meta={
                  Object {
                    "name": "type",
                    "originalProps": Object {
                      "children": Array [
                        <Option
                          value="MOTO"
                        >
                          Moto
                        </Option>,
                        <Option
                          value="CASA"
                        >
                          Casa
                        </Option>,
                        <Option
                          value="VARIE"
                        >
                          Varie
                        </Option>,
                      ],
                      "choiceTransitionName": "zoom",
                      "optionFilterProp": "children",
                      "placeholder": "Select type",
                      "prefixCls": "ant-select",
                      "showSearch": true,
                      "transitionName": "slide-up",
                    },
                    "ref": null,
                    "rules": Array [
                      Object {
                        "message": "Please select a valid product",
                        "required": true,
                      },
                    ],
                    "trigger": "onChange",
                    "validate": Array [
                      Object {
                        "rules": Array [
                          Object {
                            "message": "Please select a valid product",
                            "required": true,
                          },
                        ],
                        "trigger": Array [
                          "onChange",
                        ],
                      },
                    ],
                    "valuePropName": "value",
                  }
                }
                onKeyDown={[Function]}
                role="combobox"
                tabIndex={0}
              >
                <div
                  className="ant-select-selection__rendered"
                >
                  <div
                    className="ant-select-selection__placeholder"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "WebkitUserSelect": "none",
                        "display": "block",
                        "userSelect": "none",
                      }
                    }
                    unselectable="on"
                  >
                    Select type
                  </div>
                  <div
                    className="ant-select-search ant-select-search--inline"
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    <div
                      className="ant-select-search__field__wrap"
                    >
                      <input
                        autoComplete="off"
                        className="ant-select-search__field"
                        id="type"
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        value=""
                      />
                      <span
                        className="ant-select-search__field__mirror"
                      >
                        
                        Â 
                      </span>
                    </div>
                  </div>
                </div>
                <span
                  className="ant-select-arrow"
                  onClick={[Function]}
                  style={
                    Object {
                      "WebkitUserSelect": "none",
                      "userSelect": "none",
                    }
                  }
                  unselectable="on"
                >
                  <i
                    className="anticon anticon-down ant-select-arrow-icon"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="down"
                      fill="currentColor"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </i>
                </span>
              </div>
            </div>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <div
              className="ant-input-number"
              style={Object {}}
            >
              <div
                className="ant-input-number-handler-wrap"
              >
                <span
                  aria-disabled={false}
                  aria-label="Increase Value"
                  className="ant-input-number-handler ant-input-number-handler-up "
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  unselectable="unselectable"
                >
                  <i
                    className="anticon anticon-up ant-input-number-handler-up-inner"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="up"
                      fill="currentColor"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                      />
                    </svg>
                  </i>
                </span>
                <span
                  aria-disabled={false}
                  aria-label="Decrease Value"
                  className="ant-input-number-handler ant-input-number-handler-down "
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  unselectable="unselectable"
                >
                  <i
                    className="anticon anticon-down ant-input-number-handler-down-inner"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="down"
                      fill="currentColor"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </i>
                </span>
              </div>
              <div
                aria-valuemin={-9007199254740991}
                className="ant-input-number-input-wrap"
                role="spinbutton"
              >
                <input
                  autoComplete="off"
                  className="ant-input-number-input"
                  data-__field={
                    Object {
                      "name": "amount",
                    }
                  }
                  data-__meta={
                    Object {
                      "name": "amount",
                      "originalProps": Object {
                        "placeholder": "Insert amount: â¬",
                        "prefixCls": "ant-input-number",
                        "step": 1,
                      },
                      "ref": null,
                      "rules": Array [
                        Object {
                          "message": "Please insert a valid amount",
                          "required": true,
                        },
                      ],
                      "trigger": "onChange",
                      "validate": Array [
                        Object {
                          "rules": Array [
                            Object {
                              "message": "Please insert a valid amount",
                              "required": true,
                            },
                          ],
                          "trigger": Array [
                            "onChange",
                          ],
                        },
                      ],
                      "valuePropName": "value",
                    }
                  }
                  id="amount"
                  min={-9007199254740991}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseUp={[Function]}
                  placeholder="Insert amount: â¬"
                  required={false}
                  step={1}
                  value=""
                />
              </div>
            </div>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <textarea
              className="ant-input"
              data-__field={
                Object {
                  "name": "description",
                }
              }
              data-__meta={
                Object {
                  "name": "description",
                  "originalProps": Object {
                    "placeholder": "Insert short description",
                    "prefixCls": "ant-input",
                  },
                  "ref": null,
                  "trigger": "onChange",
                  "validate": Array [],
                  "valuePropName": "value",
                }
              }
              id="description"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Insert short description"
              style={Object {}}
              value=""
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              type="submit"
            >
              <span>
                Add
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </form>
</div>
`;
